pkg/apis/core/validation/validation.go

```diff
@@ -5764,23 +5764,27 @@ func validateNonSpecialIP(ipAddress string, fldPath *field.Path) field.ErrorList
 	// unspecified and loopback addresses are nonsensical and link-local
 	// addresses tend to be used for node-centric purposes (e.g. metadata
 	// service).
+	//
+	// IPv6 references
+	// - https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml
+	// - https://www.iana.org/assignments/ipv6-multicast-addresses/ipv6-multicast-addresses.xhtml
 	allErrs := field.ErrorList{}
 	ip := net.ParseIP(ipAddress)
 	if ip == nil {
 		allErrs = append(allErrs, field.Invalid(fldPath, ipAddress, "must be a valid IP address"))
 		return allErrs
 	}
 	if ip.IsUnspecified() {
-		allErrs = append(allErrs, field.Invalid(fldPath, ipAddress, "may not be unspecified (0.0.0.0)"))
+		allErrs = append(allErrs, field.Invalid(fldPath, ipAddress, "may not be unspecified (0.0.0.0, ::)"))
```

Comment:
Nit: This would be clearer if it output the appropriate empty values for the IP family instead of always outputting both.

Reply:
why not just output the `ip.String()` ?

Reply:
Keep both for documentation or remove and just print the ip addr and make it more maintainable?

What do people think?

Reply:
my thinking is that the person that has networking knowledge will understand both, 
however, someone with less networking experience will use a random ip, so returning the ip value only will not give him any information.
I think that documenting the ranges can benefit users and make the error more clear

Reply:
The definition of IsUnspecified() is leterally checking for the zero-values.  Printing both here is (to me) strictly worse than just printing the IP the user provided.  The ones below (IsLoopback, etc) are ranges and make more sense.  Printing family-specific errors would be best.  If you give me IPv4, you should not get an IPv6 error

Reply:
agree

Reply:
fixed